hudi-common/src/main/java/org/apache/hudi/common/model/HoodieFileGroup.java

```diff
@@ -101,12 +103,36 @@ public void addBaseFile(HoodieBaseFile dataFile) {
 
   /**
    * Add a new log file into the group.
+   *
+   * <p>CAUTION: the log file must be added in sequence of the delta commit time.
```

Comment:
lets add a validation check for this?

Reply:
I think we are cool, the `addLogFile` only has one caller.

Reply:
Filesystem view calls this while building filegroups but that call is in sycnhronized block. However, I am assuming we have a multi writer and concurrent reader test to validate. Or we can make this visible for testing and start two threads to add log files and validate.